samtranslator/model/sam_resources.py

```diff
@@ -1888,6 +1922,8 @@ def _construct_iam_policy(
         destination: ConnectorResourceReference,
         profile: ConnectorProfile,
         resource_resolver: ResourceResolver,
+        dest_index: int,
+        multi_dest: bool,
```

Comment:
Instead of these two variables, can we make `dest_index` as a optional? The checks below can become 
`policy_name = f"{self.logical_id}PolicyDestination{dest_index}" if dest_index != None else f"{self.logical_id}Policy"`

Reply:
We would still need a flag to identify it's a single destination or a list only contains one destination.